posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc

```diff
@@ -0,0 +1,118 @@
+
+Keeping in line with the app-centric design, Liberty Tools provides code assistance for both Jakarta EE and MicroProfile APIs. Code completion, hover descriptions, diagnostics and quick-fixes save time, reduce errors, and keep you focused on your code. Liberty Tools accomplishes this by integrating with the link:https://github.com/eclipse/lsp4jakarta[Language Server for Jakarta EE] and the link:https://github.com/eclipse/lsp4mp[Language Server for MicroProfile], both open source projects which are continually adding new features and support.
+
+While much of the runtime mechanics are automated with Liberty Tools, some runtime configuration is, of course, necessary. However, this process has been streamlined with config assistance for `server.xml`, `bootstrap.properties`, and `server.env` files. Rather than having to refer to the documentation to find the correct Liberty feature to add or the appropriate value to set for your specific logging attribute, Liberty Tools provides lists of options and descriptions all without having to leave Eclipse.
+
+Prefer another IDE? Liberty Tools for Eclipse IDE maintains functional parity with plugins for link:https://marketplace.visualstudio.com/items?itemName=Open-Liberty.liberty-dev-vscode-ext[Visual Studio Code] and link:https://plugins.jetbrains.com/plugin/14856-liberty-tools/[Intellij IDEA].
+
+== Requirements
+
+Liberty Tools for Eclipse IDE is designed to work with link:https://github.com/OpenLiberty/liberty-tools-eclipse/blob/main/docs/user-guide.md#software-requirements[supported versions] of both 
+the IDE for Java EE Developers and the IDE for Java Developers. While Java 17 is needed to run the tool, you can choose a different version of Java to run your application. 
+
+Since Liberty Tools uses the Liberty Maven and Liberty Gradle plugins behind the scenes, these plugins must be configured in your applications `pom.xml` or `build.gradle` files. Liberty Tools will also need to run the Maven or Gradle CLI which can be accomplished by configuring the Maven and Gradle home in the link:https://github.com/OpenLiberty/liberty-tools-eclipse/blob/main/docs/user-guide.md#setting-preferences[tool's preferences], or including a wrapper with the application.
+
+== Managing your applications
+
+Any application in your Eclipse workspace that is configured with the Liberty Maven or Liberty Gradle plugin is automatically enabled to run with Liberty Tools. Actions are available to manage these apps such as `Start`, `Stop`, `Debug` and `Run Tests`, and you can run these actions from either the Liberty Dashboard or the Eclipse "Run As" menu.
+
+=== Liberty Dashboard
+
+The Liberty Dashboard consolidates the Liberty enabled applications in your workspace into a single view. To open the dashboard, click on Open Liberty UFO icon on the top tray. To run an action, simply right click on an app in the list.
```

Comment:
on the top tray (or press Alt+L) 

Promote the keyboard shortcut ?

Reply:
Good idea